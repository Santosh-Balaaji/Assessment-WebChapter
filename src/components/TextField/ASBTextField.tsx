import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import InputLabel from '@material-ui/core/InputLabel';
import { IASBTextFieldProps } from '../../utils/interfaces';

const ASBTextField: React.FC<IASBTextFieldProps> = ({
  id,
  label,
  name,
  value,
  size,
  placeholder,
  variant,
  inlineLabel,
  onChange,
  onBlur,
  fullWidth,
  className,
}) => (
  <>
    {inlineLabel && label && <InputLabel htmlFor={label}>{label}</InputLabel>}
    <TextField
      data-testid="TextField"
      id={id}
      name={name}
      value={value}
      label={inlineLabel ? '' : label}
      placeholder={placeholder}
      variant={variant}
      size={size}
      fullWidth={fullWidth}
      onChange={onChange}
      onBlur={onBlur}
      className={className}
    />
  </>
);

ASBTextField.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  variant: PropTypes.oneOf(['filled', 'outlined', 'standard']),
  size: PropTypes.oneOf(['small', 'medium']),
  inlineLabel: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  fullWidth: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onBlur: PropTypes.func.isRequired,
  className: PropTypes.string,
};

ASBTextField.defaultProps = {
  size: 'small',
  variant: 'outlined',
  placeholder: '',
  inlineLabel: false,
  label: '',
  fullWidth: false,
  value: null,
  className: '',
};

export default ASBTextField;
