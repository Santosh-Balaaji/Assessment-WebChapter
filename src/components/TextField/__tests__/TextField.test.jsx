import React from 'react';
import { shallow } from 'enzyme';
import ASBTextField from '../ASBTextField';

describe('ASB Text Field', () => {
  let wrapper;
  const mockFnOnChange = jest.fn();
  const mockFnOnBlur = jest.fn();
  const stubProps = {
    id: 'textfield',
    label: 'TextField',
    placeholder: 'Enter',
    size: 'small',
    variant: 'filled',
    inlineLabel: false,
    name: 'textfield',
    onChange: mockFnOnChange,
    fullWidth: false,
    value: 'Hello World',
    onBlur: mockFnOnBlur,
  };

  beforeAll(() => {
    wrapper = shallow(<ASBTextField {...stubProps} />);
  });

  it('should render default component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange on Change of Event', () => {
    wrapper
      .find('WithStyles(ForwardRef(TextField))[data-testid="TextField"]')
      .simulate('change', { target: { name: 'Sample', value: '12/12/2020' } });
    expect(mockFnOnChange).toHaveBeenCalled();
  });

  it('should call handleBlur function on Blur', () => {
    wrapper
      .find('WithStyles(ForwardRef(TextField))[data-testid="TextField"]')
      .simulate('blur');
    expect(mockFnOnBlur).toHaveBeenCalled();
  });
});
