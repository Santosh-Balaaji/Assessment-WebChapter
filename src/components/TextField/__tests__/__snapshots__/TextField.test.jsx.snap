// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASB Text Field should render default component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TextField))
    className=""
    data-testid="TextField"
    fullWidth={false}
    id="textfield"
    label="TextField"
    name="textfield"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder="Enter"
    size="small"
    value="Hello World"
    variant="filled"
  />
</Fragment>
`;
